snapshot-labs/sx-monorepo
# feat: add AI text to speech

### Summary



Add a feature to listen to the proposal's body, via OpenAI text to speech.
Audio file is loaded via our custom backend (https://github.com/snapshot-labs/snapshot-sidekick/pull/244), and cached after first load

![Screenshot 2024-03-29 at 22 59 18](https://github.com/snapshot-labs/sx-monorepo/assets/495709/e1906a55-38c8-4c41-a061-4e31b343a9d2)

Eligible proposal:
- Offchain proposals
- Body length >= 1 chars
- Body length < 4096 chars


### How to test


1. Go to an eligibile proposal
2. In the proposal top-right menu, click on the PLAY button
3. Button should switch to "load" state while fetching the audio from backend
4. Once loaded, the button should switch to a PAUSE icon, and audio should start playing
5. Clicking on the button will toggle between PLAY and PAUSE state for the audio and the icon
6. At the end of the audio, audio should stop and button should revert to a PLAY icon
7. You can click on the button to play the audio again from the start
8. When audio is playing, it should stop playing when navigating to another page
9. When audio is loading, it should not play after loading of you navigated to another page before audio finished loading

### To-Do

- [x] Waiting for https://github.com/snapshot-labs/snapshot-sidekick/pull/244
